static/src/javascripts/projects/common/modules/identity/api.js

```diff
@@ -0,0 +1,284 @@
+ */
+export const getUserFromCookie = (): ?Object => {
+    if (userFromCookieCache === null) {
+        const cookieData = getCookieByName(Id.cookieName);
+        let userData = null;
+
+        if (cookieData) {
+            userData = JSON.parse(decodeBase64(cookieData.split('.')[0]));
+        }
+        if (userData) {
+            const displayName = decodeURIComponent(userData[2]);
+            userFromCookieCache = {
+                id: userData[0],
+                primaryEmailAddress: userData[1], // not sure where this is stored now - not in the cookie any more
+                displayName,
+                accountCreatedDate: userData[6],
+                emailVerified: userData[7],
+                rawResponse: cookieData,
+            };
+        }
+    }
+
+    return userFromCookieCache;
+};
+
+/**
+ * @return {boolean}
+ */
+export const isUserLoggedIn = () => getUserFromCookie() !== null;
+
+/**
+ * Gets the currently logged in user data from the identity api
+ * @param {function} callback
+ */
+export const getUserFromApi = mergeCalls(mergingCallback => {
+    const apiRoot = Id.idApiRoot ? Id.idApiRoot : '';
+
+    if (isUserLoggedIn()) {
+        ajax({
+            url: `${apiRoot}/user/me`,
+            type: 'jsonp',
+            crossOrigin: true,
+        }).then(response => {
+            if (response.status === 'ok') {
+                mergingCallback(response.user);
+            } else {
+                mergingCallback(null);
+            }
+        });
+    } else {
+        mergingCallback(null);
+    }
+});
+
+/**
+ * Clears the caches and state, primarily for testing.
+ */
+export const reset = () => {
+    getUserFromApi.reset();
+    userFromCookieCache = null;
+};
+
+/**
+ * @return {string}
+ */
+export const getCookie = () => getCookieByName(Id.cookieName);
+
+/**
+ * @return {string}
+ */
+export const getUrl = () => config.page.idUrl;
+
+/**
+ * Gets the currently logged in user data from the identity api and
+ * refreshes the users cookie at the same time.
+ */
+export const getUserFromApiWithRefreshedCookie = () => {
+    const endpoint = '/user/me';
+    const request = ajax({
+        url: Id.idApiRoot ? Id.idApiRoot + endpoint : endpoint,
+        type: 'jsonp',
+        data: {
+            refreshCookie: true,
+        },
+    });
+
+    return request;
+};
+
+/**
+ * Wrap window.location.href so it can be spied in unit tests
+ */
+export const redirectTo = (url: string): void => {
+    window.location.href = url;
+};
+
+/**
+ * Returns user object when signed in, otherwise redirects to sign in with configurable absolute returnUrl
+ */
+export const getUserOrSignIn = (paramUrl: ?string): ?Object => {
+    let returnUrl = paramUrl;
+
+    if (isUserLoggedIn()) {
+        return getUserFromCookie();
+    }
+
+    returnUrl = encodeURIComponent(returnUrl || document.location.href);
+    const url = `${getUrl() ? getUrl() : ''}/signin?returnUrl=${returnUrl}`;
```

Comment:
We can remove the ternary `getUrl() ? getUrl() : ''` from this line and replace with `getUrl() || ''`. By doing this we're only calling `getUrl` once, rather than potentially calling it twice as we are now.